lib/foreman/renderer.rb

```diff
@@ -27,26 +29,36 @@ def template_logger
     end
 
     def host_enc(*path)
+      check_host
```

Comment:
Shouldn't the `host_enc` also have the `host_enc!`, `host_enc_true?` and `host_enc_false?`

Reply:
sorry it took me a while,

`host_enc` already raises and exception (KeyError) if it can't fetch a given parameter. Removing it would break existing API so I just improved it so it raises more appropriate exception that inherits from `RenderingError`

`host_enc_true?` and `host_enc_false?` should not be required since these values keep their types, if smart class parameter or smart variable is defined as Boolean, it will be always caster to true or false.